ERCS/erc-7730.md

```diff
@@ -0,0 +1,442 @@
+
+## Motivation
+
+Properly validating a transaction on a Hardware Wallets screen (also known as Clear Signing) is a key element of good security practices for end users when interacting with any Blockchain. Unfortunately, most data to sign, even enriched with the data structure description (ABI or EIP-712 types) are not self-sufficient in term of correctly displaying them to the user for review. Among other things:
+
+* Function name or main message type is often a convoluted name and does not translate to a clear intent for the user
+* Fields in the data to sign are tied to primitive types only, but those can be displayed many different ways. For instance, integers can be displayed as percentages, dates, etc...
+* Some fields require additional metadata to by displayed correctly, for instance token amounts requires knowledge of the magnitude and the ticker, as well as where to find the token address itself to be correctly formatted.
+
+This specification intends to provide a simple, open standard format to provide wallets with the additional information required to properly format a structured data to sign for review by users.
+
+Providing this additional formatting information requires deep knowledge of the way a smartcontract or message is going to be used, it is expected that DApps developers will be the best placed to write such a file. The intent of an open standard is to write this file only once and have it work with most wallets supporting this standard.
+
+> Deployment model options:
+> - In each dapps repository (discoverability?)
+> - Foundation operated repository
+> - Ledger repository
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### Simple example
+
+The following is a simple example of how to clear sign a `transfer` function call on an ERC20 contract.
+
+```json
+{
+    "$schema": "https://github.com/LedgerHQ/clear-signing-format-spec/blob/master/schemas/clear-signing-schema.json",
+
+    "context": {
+        "$id": "Tether USD",
+        "contract" : {
+            "chainId": 1,
+            "address": "0xdAC17F958D2ee523a2206206994597C13D831ec7",
+            "abi": "https://api.etherscan.io/api?module=contract&action=getabi&address=0xdac17f958d2ee523a2206206994597c13d831ec7"
+        }
+    }, 
+
+    "metadata": {
+        "owner": "Tether",
+        "info": {
+            "legalName": "Tether Limited",
+            "url": "https://tether.to/",
+            "lastUpdate": "2017-11-28T12:41:21Z"  
+        }
+    },
+
+    "display": {
+        "formats": {
+            "0xa9059cbb": {
```

Comment:
Is there a benefit in using the 4-byte signature as opposed to using the function name i.e. `transfer` or `transfer(address,uint256)`?

Reply:
You are right that using the function signature would be more readable! We can't use just the function name since we need unicity of those keys, but the full signature used for selector computation would work. Will do an update to allow for both selectors and function signatures